pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -39,15 +64,39 @@ type provider interface {
 type Service struct {
 	outboundTransport transport.OutboundTransport
 	store             storage.Store
+	callbackChannel   chan didCommChMessage
+	actionEvent       chan<- event.DIDCommEvent
+	streamEvents      []chan<- dispatcher.DIDCommMsg
+	lock              sync.RWMutex
+	streamLock        sync.RWMutex
+	execute           bool
 }
 
 // New return didexchange service
 func New(store storage.Store, prov provider) *Service {
-	return &Service{outboundTransport: prov.OutboundTransport(), store: store}
+	svc := &Service{
+		// TODO Outbound dispatcher - https://github.com/hyperledger/aries-framework-go/issues/259
+		outboundTransport: prov.OutboundTransport(),
+		store:             store,
+		// TODO channel size - https://github.com/hyperledger/aries-framework-go/issues/246
+		callbackChannel: make(chan didCommChMessage, 10),
+		// set execute to false. Consumers have to enable this by setting either RegisterActionEvent() or
+		// RegisterAutoExecute()
+		execute: false,
+	}
+
+	svc.startInternalListener()
```

Comment:
Suggest rename to "startListeningForIncomingCallbacks". At least it makes the intent more clear.

Reply:
@troyronda thoughts on this?

Reply:
I think we need to do further changes as a new PR. It's getting hard to follow.